hedera-node/docs/dynamic_address_book.md

```diff
@@ -0,0 +1,359 @@
+| `gossip_ca_certificate` | The new certificate used to sign gossip events. |
+| `nodeId` | The unique identifier of the consensus node. |
+| `weight` | The new consensus weight applied to the node. |
+
+---
+
+## Deploy a Hiero Network from Genesis
+
+### 1. Pre-Genesis State
+
+Before **genesis** (i.e., before the network starts for the first time):
+
+- Both the **Node Store** and **Roster Store** are empty or `null`.
+
+---
+
+### 2. Genesis Initialization
+
+At **network genesis**,when the network first starts up, the system initializes both node and roster stores by reading from a predefined configuration file (ex: genesis-network.json) found in `data/config`. 
+
+**The initialization process:**
+
+- Loads each node’s properties from `data/config/genesis-network.json` into the **Node Store** and **Roster Store**
+
+**Example configuration file: genesis-network.json (base64 encoded)**
+
+```json
+{
+  "nodeMetadata": [
+    {
+      "node": {
+        "accountId": {
+          "accountNum": "3"
+        },
+        "adminKey": {
+          "ed25519": "1uewl5ybyTYuWew23yDKgzAmcgSfj3BG/wvMCNlYHg0="
+        },
+        "description": "node 0",
+        "gossipCaCertificate": "MIIDyTCCAjGgAwIBAgIIY2692yMBcSwwDQYJKoZIhvcNAQEMBQAwEjEQMA4GA1UEAxMHcy1ub2RlMTAgFw0yNDAyMDgwMzAyNTVaGA8yMTI0MDE...",
+        "gossipEndpoint": [
+          {
+            "ipAddressV4": "wKgBCg==",
+            "port": 50111
+          },
+          {
+            "ipAddressV4": "LMk3EQ==",
+            "port": 50111
+          }
+        ],
+        "grpcCertificateHash": "hqwSJJ9e1Z9sIsVGl40IQTAGhW1nLjmAfB1prtltt9WFM/NXWOCJc9PRSO1Pi757...",
+        "grpcProxyEndpoint": {
+          "domainName": "grpc-proxy.test.com",
+          "port": 443
+        },
+        "nodeId": "0",
+        "serviceEndpoint": [
+          {
+            "ipAddressV4": "I+3ItA==",
+            "port": 50211
+          },
+          {
+            "ipAddressV4": "rBD+Aw==",
+            "port": 50212
+          }
+        ],
+        "weight": "10"
+      },
+      "rosterEntry": {
+        "gossipCaCertificate": "MIIDyTCCAjGgAwIBAgIIY2692yMBcSwwDQ...",
+        "gossipEndpoint": [
+          {
+            "ipAddressV4": "wKgBCg==",
+            "port": 50111
+          },
+          {
+            "ipAddressV4": "LMk3EQ==",
+            "port": 50111
+          }
+        ],
+        "nodeId": "0",
+        "weight": "10"
+      }
+    }
+  ]
+}
+```
+
+---
+
+### 3. Post-Genesis Updates
+
+Once the network is operational, node details can change through NodeCreateTransaction, NodeUpdateTransaction, and NodeDeleteTransaction.
+
+**What happens when updates occur?**
+
+- The **Node Store** updates in-place for all changed fields (`service_endpoints`, `admin_key`, etc.).
+- The **Roster Store** updates only if gossip-related fields are modified.
+    - Contains current roster and candidate roster
+- The **Candidate Roster** - a new roster that may be adopted at the next network upgrade.
```

Comment:
need to verify if it is only the candidate roster that updates gossip values and the current roster remains unchanged. Following need to update the example accordingly